client/web/src/search/suggestion/SmartSearchPreview.tsx

```diff
@@ -0,0 +1,176 @@
+        }, [query])
+    )
+
+    useEffect(() => {
+        if (results?.alert?.proposedQueries) {
+            const resultNum: number = results.alert.proposedQueries.reduce(
+                (acc: number, proposedQuery: ProposedQuery): number => {
+                    let proposedQueryResultCount = 0
+                    const proposedQueryResultCountGroup = proposedQuery.annotations?.filter(
+                        ({ name }) => name === 'ResultCount'
+                    )
+                    proposedQueryResultCountGroup?.forEach(
+                        r => (proposedQueryResultCount += Number(r.value.replace(/\D/g, '')))
+                    )
+                    acc += proposedQueryResultCount
+                    return acc
+                },
+                0
+            )
+
+            // SmartSearch count stops after 500
+            setResultNumber(resultNum > 500 ? '500+' : resultNum)
+        }
+        return
+    }, [results])
+
+    return (
+        <>
+            {results?.state === 'loading' ? (
+                <div className="mb-5">
+                    <H2 as={H3}>Please wait. Smart Search is trying variations on your query...</H2>
+
+                    <div className={classNames(styles.shimmerContainer, 'rounded my-3 col-6')}>
+                        <div className={classNames(styles.shimmerAnimate, 'absolute top-0 overflow-hidden')} />
+                    </div>
+
+                    <div className={classNames(styles.shimmerContainer, 'rounded mb-3 col-4')}>
+                        <div className={classNames(styles.shimmerAnimateSlower, 'absolute top-0 overflow-hidden')} />
+                    </div>
+
+                    <EnableSmartSearch
+                        query={query}
+                        history={history}
+                        patternType={patternType}
+                        caseSensitive={caseSensitive}
+                    />
+                </div>
+            ) : results?.state === 'complete' && !!results?.alert?.proposedQueries ? (
+                <div className="mb-5">
+                    <H2 as={H3}>However, Smart Smart found {resultNumber} results:</H2>
+
+                    <ul className={classNames(styles.container, 'px-0 mb-3')}>
+                        {results?.alert?.proposedQueries?.map(item => (
+                            <li key={item.query}>
+                                <Link
+                                    to={createLinkUrl({
+                                        pathname: '/search',
+                                        search: formatSearchParameters(new URLSearchParams({ q: item.query })),
+                                    })}
+                                    className={classNames(styles.link, 'px-0')}
+                                >
+                                    <span className="p-1 bg-code">
+                                        <SyntaxHighlightedSearchQuery
+                                            query={query}
```

Comment:
The list is currently showing the current query instead of the proposed query, I think this would fix it?

```suggestion
                                            query={item.query}
```

Reply:
Good catch, thanks!